science/mmtf-cpp/Portfile

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.0
+PortGroup           github 1.0
+
+github.setup        rcsb mmtf-cpp 1.0.0 v
+name                mmtf-cpp
+
+categories          science
+platforms           darwin
+maintainers         {gmail.com:howarth.at.macports @jwhowarth} openmaintainer
+license             MIT
+
+description         MMTF in C++
+long_description    The macromolecular transmission format (MMTF) is a binary encoding of biological structures.
+homepage            https://github.com/rcsb/mmtf-cpp
+
+checksums           rmd160  ddee6c5f449254bd58f1a919bfe7e8818e77a026 \
+                    sha256  276e19f1019b080969d40b85094ff604dc2f1762707be0fa4c59bf67b8cff2e5 \
+                    size    59370
+
```

Comment:
please use MacPorts version of the MessagePack C++ library (version 2.1.5 or newer) instead of the vendored version in the repository. You can to add `depends_lib-append  port:msgpack` as the port is already present.

Reply:
This package don't build any binaries. It only installs...

`drwxr-xr-x  0 root   wheel       0 May 23 17:09 ./
-rw-r--r--  0 root   wheel     287 May 23 17:09 ./+STATE
-rw-r--r--  0 root   wheel     849 May 23 17:09 ./+PORTFILE
-rw-r--r--  0 root   wheel    1356 May 23 17:09 ./+CONTENTS
-rw-r--r--  0 root   wheel     132 May 23 17:09 ./+DESC
-rw-r--r--  0 root   wheel      12 May 23 17:09 ./+COMMENT
drwxr-xr-x  0 root   wheel       0 May 23 17:09 ./opt/
drwxr-xr-x  0 root   wheel       0 May 23 17:09 ./opt/local/
drwxr-xr-x  0 root   wheel       0 May 23 17:09 ./opt/local/include/
-rw-r--r--  0 root   wheel     564 Feb  5  2019 ./opt/local/include/mmtf.hpp
drwxr-xr-x  0 root   wheel       0 May 23 17:09 ./opt/local/include/mmtf/
-rw-r--r--  0 root   wheel   16181 Feb  5  2019 ./opt/local/include/mmtf/binary_decoder.hpp
-rw-r--r--  0 root   wheel     993 Feb  5  2019 ./opt/local/include/mmtf/errors.hpp
-rw-r--r--  0 root   wheel    8910 Feb  5  2019 ./opt/local/include/mmtf/encoder.hpp
-rw-r--r--  0 root   wheel   10730 Feb  5  2019 ./opt/local/include/mmtf/binary_encoder.hpp
-rw-r--r--  0 root   wheel    2873 Feb  5  2019 ./opt/local/include/mmtf/decoder.hpp
-rw-r--r--  0 root   wheel   27538 Feb  5  2019 ./opt/local/include/mmtf/structure_data.hpp
-rw-r--r--  0 root   wheel    3727 Feb  5  2019 ./opt/local/include/mmtf/export_helpers.hpp
-rw-r--r--  0 root   wheel    6314 Feb  5  2019 ./opt/local/include/mmtf/msgpack_decoder.hpp
-rw-r--r--  0 root   wheel    7793 Feb  5  2019 ./opt/local/include/mmtf/map_decoder.hpp
-rw-r--r--  0 root   wheel    2410 Feb  5  2019 ./opt/local/include/mmtf/object_encoders.hpp`

Reply:
upstream says: "You need the headers of the MessagePack C++ library (version 2.1.5 or newer)."

Also, as you can see below, the headers include `<msgpack.hpp>`:
```
> grep include `port contents mmtf-cpp | grep opt` | grep msgpack
/opt/local/include/mmtf/binary_decoder.hpp:#include <msgpack.hpp>
/opt/local/include/mmtf/decoder.hpp:#include "msgpack_decoder.hpp"
/opt/local/include/mmtf/decoder.hpp:#include <msgpack.hpp>
/opt/local/include/mmtf/map_decoder.hpp:#include <msgpack.hpp>
/opt/local/include/mmtf/msgpack_decoder.hpp:#include "structure_data.hpp"
/opt/local/include/mmtf/msgpack_decoder.hpp:#include "map_decoder.hpp"
/opt/local/include/mmtf/msgpack_decoder.hpp:#include "errors.hpp"
/opt/local/include/mmtf/msgpack_decoder.hpp:#include <msgpack.hpp>
/opt/local/include/mmtf/object_encoders.hpp:#include <msgpack.hpp>
/opt/local/include/mmtf/structure_data.hpp:#include <msgpack.hpp>
```
and that file comes from the `msgpack` port:
```
> port provides /opt/local/include/msgpack.hpp
/opt/local/include/msgpack.hpp is provided by: msgpack
```